REGRESSION (270199@main): [visionOS] YouTube video fullscreen layout does not fill screen to fit the full window

pxlcoder opened the pull request:
#### 0d56bc1e302098ab1649ceda9e2390d3a625f787
<pre>
REGRESSION (270199@main): [visionOS] YouTube video fullscreen layout does not fill screen to fit the full window
<a href="https://bugs.webkit.org/show_bug.cgi?id=267135">https://bugs.webkit.org/show_bug.cgi?id=267135</a>
<a href="https://rdar.apple.com/119830067">rdar://119830067</a>

Reviewed by Richard Robinson and Wenson Hsieh.

270199@main removed two pieces of logic that previously guaranteed that the
layout parameters (such as `window.innerWidth`) exposed to the web during the
&quot;fullscreenchange&quot; event reflected the fullscreen size.

Prior to that change `[webView setFrame:];` and
`[webView _overrideLayoutParametersWithMinimumLayoutSize:maximumUnobscuredSizeOverride:];`
in `-[WKFullScreenWindowController enterFullScreen:]` guaranteed that behavior.

YouTube sets the video element&apos;s width/height to `window.innerWidth`/`window.innerHeight`
in the &quot;fullscreenchange&quot; event, resulting in video that doesn&apos;t fill the screen.

Even when the &quot;resize&quot; event is fired, a flash of the incorrect size is observed.

Fix by restoring the call to `[webView setFrame:];`.
`[webView _overrideLayoutParametersWithMinimumLayoutSize:maximumUnobscuredSizeOverride:];`
is not restored since the goal in 270199@main to avoid overriding viewport layout
parameters remains.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen:]):

Canonical link: <a href="https://commits.webkit.org/272752@main">https://commits.webkit.org/272752@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d31544248acbc3f818c94e5c9217812f725521d9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32945 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11705 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34828 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35573 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29763 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14046 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8881 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29207 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33405 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9868 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29430 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8602 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/8750 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36905 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29925 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29796 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34883 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8876 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6832 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32738 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10570 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7639 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9480 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9505 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
```diff
@@ -953,6 +953,10 @@ - (void)enterFullScreen:(CGSize)videoDimensions
 
         [webView setAutoresizingMask:(UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight)];
+        // FIXME (267139): Changing the web view's frame should ideally only be done by the fullscreen view controller.
+        // However, this adjustment is currently necssary to ensure that the layout parameters exposed to the web
```
Nit - spelling

```suggestion
        // However, this adjustment is currently necessary to ensure that the layout parameters exposed to the web
```

aprotyas reviewed (commented):
LGTM with the typo fix suggested by Wenson

pxlcoder committed:
REGRESSION (270199@main): [visionOS] YouTube video fullscreen layout does not fill screen to fit the full window
https://bugs.webkit.org/show_bug.cgi?id=267135
rdar://119830067

Reviewed by Richard Robinson and Wenson Hsieh.

270199@main removed two pieces of logic that previously guaranteed that the
layout parameters (such as `window.innerWidth`) exposed to the web during the
"fullscreenchange" event reflected the fullscreen size.

Prior to that change `[webView setFrame:];` and
`[webView _overrideLayoutParametersWithMinimumLayoutSize:maximumUnobscuredSizeOverride:];`
in `-[WKFullScreenWindowController enterFullScreen:]` guaranteed that behavior.

YouTube sets the video element's width/height to `window.innerWidth`/`window.innerHeight`
in the "fullscreenchange" event, resulting in video that doesn't fill the screen.

Even when the "resize" event is fired, a flash of the incorrect size is observed.

Fix by restoring the call to `[webView setFrame:];`.
`[webView _overrideLayoutParametersWithMinimumLayoutSize:maximumUnobscuredSizeOverride:];`
is not restored since the goal in 270199@main to avoid overriding viewport layout
parameters remains.

* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController enterFullScreen:]):

Canonical link: https://commits.webkit.org/272752@main
